environments/puppet/hiera.yaml

```diff
@@ -23,16 +56,41 @@ hierarchy:
     - "%{facts.os.name}/%{facts.os.release.major}.yaml"
     - "%{facts.os.name}.yaml"
 
+  - name: EYAML per-hostgroup data
+    lookup_key: eyaml_lookup_key
+    paths:
+    - "hostgroups/%{hostgroup}.eyaml"
```

Comment:
This is a top-scope variable, so [it should have a `::` prefix](https://puppet.com/docs/puppet/latest/hiera_merging.html#interpolate_puppet_variable).

```suggestion
    - "hostgroups/%{::hostgroup}.eyaml"
```